docs/docs/algorithms/cycle-detection/dfs-based.md

```diff
@@ -2,4 +2,55 @@
 sidebar_position: 1
 ---
 
-# DFS Based Cycle Detection
\ No newline at end of file
+# DFS Based Cycle Detection
+A DFS based cycle detection algorithm is used to identify cycles in graphs, both directed and undirected.
+The algorithm doesn't work on edges with weight and therefore cannot be used for negative weight cycles.
+
```

Comment:
[minor]: It sounds like this is a disadvantage of the algorithm, but I am wondering if negative weight cycles would ever influence cycle detection. I would argue the algorithm does work for weighted graphs, but the edge weight will simply not be considered 🙈 

What do you think about this?

```suggestion
The algorithm can be used to detect cycles in the structure of a graph, as it does not consider edge weights.

```

Reply:
I agree that your suggestion is more correct and a bit more readable.